test/into_variant_test.cpp

```diff
@@ -0,0 +1,111 @@
+      vt_t<vt_t<int>>>);
+  static_assert(is_same_v<decltype(snd1)::error_types<variant>,
+      variant<std::exception_ptr>>);
+  static_assert(decltype(snd1)::sends_done == false);
+
+  auto snd2 = just_error(42)
+            | into_variant();
+  static_assert(is_same_v<decltype(snd2)::value_types<variant, tuple>,
+      vt_t<variant<>>>);
+  static_assert(is_same_v<decltype(snd2)::error_types<variant>,
+      variant<int>>);
+  static_assert(decltype(snd2)::sends_done == false);
+
+  auto snd3 = just_done()
+            | into_variant();
+  static_assert(is_same_v<decltype(snd3)::value_types<variant, tuple>,
+      vt_t<variant<>>>);
+  static_assert(is_same_v<decltype(snd3)::error_types<variant>,
+      variant<>>);
+  static_assert(decltype(snd3)::sends_done == true);
+
+  auto snd4 = when_all( just(42), just(42.0), just_error(42) )
+            | into_variant();
+  static_assert(is_same_v<decltype(snd4)::value_types<variant, tuple>,
+      vt_t<vt_t<vt_t<int>, vt_t<double>, variant<>>>>);
+  static_assert(is_same_v<decltype(snd4)::error_types<variant>,
+      variant<std::exception_ptr, int>>);
+  static_assert(decltype(snd4)::sends_done == true);
+
+  auto snd5 = just(42) | let_done([]{return just("hello");}) | into_variant();
+  static_assert(is_same_v<decltype(snd5)::value_types<variant, tuple>,
+      vt_t<variant<tuple<int>, tuple<const char*>>>>);
+  static_assert(is_same_v<decltype(snd5)::error_types<variant>,
+      variant<std::exception_ptr>>);
+  static_assert(decltype(snd5)::sends_done == false);
+}
+
+TEST(IntoVariant, Working){
+  bool called = false;
+  auto x = sync_wait(into_variant(when_all(
+          just(42),
+          just(42.0) | then([&](double d){
+            called = true;
+            return d + 1;
+          }))));
+  EXPECT_TRUE(called);
+  EXPECT_TRUE(x.has_value());
+  const auto& [vt_first_val, vt_second_val] = std::get<0>(x.value());
+  const auto& [first_val] = std::get<0>(vt_first_val);
+  const auto [second_val] = std::get<0>(vt_second_val);
+  EXPECT_EQ(first_val, 42);
+  EXPECT_EQ(second_val, 43.0);
+}
+
+TEST(IntoVariant, Pipeable){
+  bool called = false;
+  auto x = when_all(
+      just(42),
+      just(42.5)
+      | then([&](double d){
+          called = true;
+          return d + 1;
+        }))
+      | into_variant()
+      | sync_wait();
+  EXPECT_TRUE(called);
+  EXPECT_TRUE(x.has_value());
+  const auto& [vt_first_val, vt_second_val] = std::get<0>(x.value());
+  const auto& [first_val] = std::get<0>(vt_first_val);
+  const auto [second_val] = std::get<0>(vt_second_val);
+  EXPECT_EQ(first_val, 42);
+  EXPECT_EQ(second_val, 43.5);
+}
+
+TEST(IntoVariant, OneOfPossibleValues){
+  bool called = false;
+  auto x = just(42)
+    | let_done([&]{ called = true; return just(42.5); }) 
+    | into_variant()
+    | sync_wait();
+  EXPECT_FALSE(called);
+  EXPECT_TRUE(x.has_value());
+  const auto& [val] = std::get<0>(x.value());
+  EXPECT_EQ(val, 42);
```

Comment:
I wanted to add ``EXPECT_ANY_THROWS(std::get<1>(x.value()));`` to signify that there was only one value in inner variant. However, this breaks the msvc build as [[nodiscard]] warning was treated as error for ``std::get<1>(x.value())``... Hence I removed that assert....